src/Microsoft.ML.AutoML/AutoMLExperiment/IMonitor.cs

```diff
@@ -30,11 +32,14 @@ internal class MLContextMonitor : IMonitor
         private readonly IChannel _logger;
         private readonly List<TrialResult> _completedTrials;
         private readonly SweepablePipeline _pipeline;
-        public MLContextMonitor(IChannel logger, SweepablePipeline pipeline)
+        public int ResourceUsageCheckInterval { get; private set; }
+
+        public MLContextMonitor(IChannel logger, SweepablePipeline pipeline, int resourceUsageCheckInterval = 5000)
```

Comment:
I'm hesitant to add `ResourceUsageCheckInterval` and `ReportTrialResourceUsage` in `IMonitor` because not all implementations of `IMonitor` need that. Moving them to `IPerformanceMonitor` might be a better way

Reply:
I agree that logically `IPerformanceMonitor` would be a better place for these and I started to make the modifications that you suggested, but how would end user access the event that the `DefaultPerformanceMonitor` fires? 
`AutoMLExperiment` doesn't have a setter for `IPerformanceMonitor` like it does for `IMonitor`, or does it?

I'm not sure if you can make changes to this PR, but feel free to make the modifications that you feel fit, I would be interested in your solution!

Reply:
I left a PR in your repo for reference. In general, that PR allows you to provide your own `IPerformanceMonitor` with a customized `ReportTrialResourceUsage` handler. Let me know if that change can resolve the issue you have